### DISCUSSION TITLE
Please add configuration options for generation of ERD with include or exclude Views & M-Views

### DISCUSSION THREAD

**Discussion by YuriDude** [dbeaver/dbeaver#14118] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
Current behavior: when an ERD is generated it includes views. 
Technically, views are **NOT** data objects, but _**derived**_ objects;  they a re just statements with Data Dictionary.
Thus, by default , they should be precluded from ERD.
Behavior for Materialized views is not checked, but it should have the same approach (excluded by default).

Furthermore: having views and/or m-views adds unnecessarily clutter on ERD which makes it less presentable to non-technical folks and somewhat confusing to developers.

**Describe the solution you'd like**
Have an option , either via global configurations or separate configuration screen or check boxes (YES/NO) for include or exclude generation of Views and/or Materialized Views in ERD.

**Describe alternatives you've considered**
None alternatives are present. Only alternative is **manually** delete views from ERD, which on large Database(s) of several hundreds (or a thousand) data objects can be cumbersome.

**Comment by uslss** [comment, marked as answer]
Hi, You could uncheck Show views option in Preferences
![image](https://user-images.githubusercontent.com/31996417/134399565-c6c8997f-46a2-4dac-ab41-14574c0a9a87.png)

**Comment by YuriDude** [comment]
@uslss - Thanks for the tip!
it did remove Views from the ERD canvass; **does it the same for Materialized Views**?

As a side note (_since i linked this "feature request" to the bug_ - https://github.com/dbeaver/dbeaver/issues/13969 ):
ERD still failed to be generated.

**Comment by uslss** [comment]
Yes, this option also removes Materialized views from ERD
